AutoCollection/ClientRequests.ts

```diff
@@ -20,6 +20,8 @@ class AutoCollectClientRequests {
 
     public static INSTANCE: AutoCollectClientRequests;
 
+    private static requestNumber = 1;
```

Comment:
you should not assume that parentId is indeed unique. It could not be unique because of
- bug on the sender side
- redirects (headers are copied as it on redirects)
- some proxy that retries requests with the same headers

The [spec suggests to append](https://github.com/lmolkova/correlation/blob/master/hierarchical_request_id.md#incoming-request) small e.g. 32-bits random integer (hex encoded to reduce length) for the parent id